omod/src/main/java/org/openmrs/module/coreapps/fragment/controller/visit/ParserEncounterIntoSimpleObjects.java

```diff
@@ -125,6 +129,9 @@ public int compare(SimpleObject o1, SimpleObject o2) {
 	}
 	
 	private SimpleObject parseObs(Obs obs, Locale locale) {
+        if(obs.getConcept() != null && obs.getConcept().isNumeric()){
```

Comment:
According to [this](https://github.com/openmrs/openmrs-core/blob/2.4.0/api/src/main/java/org/openmrs/Concept.java#L1337), `obs.getConcept().isNumeric()` will always return `false` when the return value of getConcept() is of class `Concept`.
And according to [this](https://github.com/openmrs/openmrs-core/blob/2.4.0/api/src/main/java/org/openmrs/ConceptNumeric.java#L196), the only time that `obs.getConcept().isNumeric()` will return `true` is when the return value of getConcept() is of class `ConceptNumeric` and the data type is `Numeric`.
So i do not see the value of the next line of code below.

Reply:
Hello @dkayiwa I understand that but can you see this example: 
![instance_of_numeric](https://user-images.githubusercontent.com/72515287/127285079-749b8129-7629-46a3-9ce8-86976041127d.gif)

Reply:
When i stop at a breakpoint, the `obs.getConcept().isNumeric()` is `true` but `obs.getConcept() instanceof ConceptNumeric` is `false`.

Reply:
Which OpenMRS platform version are you running?

Reply:
I am using `<openMRSVersion>2.3.3-SNAPSHOT</openMRSVersion>`.

Reply:
Instead of using a snapshot version of the platform, do you mind trying a released tag like `2.3.3` and see if you can still reproduce the problem?

Reply:
With version `2.3.3` nothing change, I can still reproduce the problem.

Reply:
> but `obs.getConcept() instanceof ConceptNumeric` is `false`.

Then it must be a hibernate proxy. Does `HibernateUtil.getRealObjectFromProxy()` help?

Reply:
@dkayiwa if i do `HibernateUtil.getRealObjectFromProxy(obs.getConcept()) instanceof ConceptNumeric` it is true. So the problem should be when i get the concept via Hibernate ( maybe lazy load) ?

Reply:
Assign the concept property to the return value of HibernateUtil.getRealObjectFromProxy